source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/FluentIconUtils.kt

```diff
@@ -0,0 +1,151 @@
+package io.adaptivecards.renderer
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Matrix
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import androidx.annotation.WorkerThread
+import com.caverock.androidsvg.SVG
+import io.adaptivecards.renderer.http.HttpRequestHelper
+import io.adaptivecards.renderer.http.HttpRequestResult
+import org.json.JSONObject
+import java.nio.charset.StandardCharsets
+
+object FluentIconUtils {
+    private const val FILLED_STYLE = "filled"
+    private const val REGULAR_STYLE = "regular"
+    private const val FLIP_IN_RTL_PROPERTY = "flipInRtl"
+
+    @WorkerThread
+    fun getFluentIcon(
+        context: Context,
+        svgURL: String,
+        iconColor: String,
+        targetIconSize: Long,
+        isFilledStyle: Boolean,
+        isRTL: Boolean,
+        callback: (drawable: Drawable?) -> Unit
+    ) {
+        val requestResult = fetchIconInfo(svgURL)
+        val iconResponse = processResponseAndRenderFluentIcon(requestResult, context, iconColor, targetIconSize, isFilledStyle)
+        iconResponse.drawable?.let {
+            if (isRTL) {
+                val flippedDrawable = flipDrawableHorizontally(it, context)
+                callback(flippedDrawable)
+            } else {
+                callback(it)
+            }
```

Comment:
Can be simplified to callback(if (isRTL) flipDrawableHorizontally(it, context) else it)